src/Core/BinCodec.fs

```diff
@@ -0,0 +1,56 @@
+﻿namespace B2R2
+
+/// Little Endian Base 128 Codec
+module LEB128 =
+  [<Literal>]
+  let max32LEB128Length = 5
+  [<Literal>]
+  let max64LEB128Length = 10
+
+  let inline private decodeUnsignedInt (bytes: byte []) (castValue: byte -> 'T) maxLength =
+    let rec decodeLoop offset value currentByte length =
+      let newValue = value ||| (castValue (currentByte &&& 0x7fuy) <<< (offset * 7))
+      if currentByte &&& 0x80uy = 0uy || offset = length - 1 then
```

Comment:
If the `bytes.length` is greater than `maxLen`, and if hit the second condition `offset = length - 1` while `currentByte &&& 0x80uy <> 0uy`, this should be considered as an error case I believe. AFAIK, the most significant byte should always be `0x80uy`, and if the original number doesn't fit the 32-bit width, we should raise an exception.

Reply:
You're right, I don't know how I missed that, the bug is fixed now, Thanks for being patient with me!